src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -2638,5 +2638,270 @@ static void verifyArray(TypeSymbol type)
                 Assert.Equal(3, array.Sizes[1]);
             }
         }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_01()
+        {
+            var source = @"
+partial class C
+{
+    public partial int M();
+    public partial long M() => 42; // 1
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (5,25): error CS8817: Both partial method declarations must have the same return type.
+                //     public partial long M() => 42; // 1
+                Diagnostic(ErrorCode.ERR_PartialMethodReturnTypeDifference, "M").WithLocation(5, 25));
+        }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_02()
+        {
+            var source = @"
+partial class C
+{
+    public partial long M();
+    public partial int M() => 42; // 1
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (5,24): error CS8817: Both partial method declarations must have the same return type.
+                //     public partial int M() => 42; // 1
+                Diagnostic(ErrorCode.ERR_PartialMethodReturnTypeDifference, "M").WithLocation(5, 24));
+        }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_03()
+        {
+            var source = @"
+using System.Collections.Generic;
+#nullable enable
+
+partial class C
+{
+    public partial string? M1();
+    public partial string M1() => ""hello"";
+
+    public partial IEnumerable<string?> M2();
+    public partial IEnumerable<string> M2() => null!;
```

Comment:
Shouldn't the difference in nullability be an error? Perhaps we could allow differences when one annotated partial is defined in a `#nullable enable` context and the other unannotated partial is defined in `#nullable disable` but allowing other cases seems incorrect to me.

Reply:
The partial method checks on parameters allow for "safe" nullable variance.

```cs
partial void M(string s);
partial void M(string? s) { } // ok

partial void M(string? s);
partial void M(string s) { } // warning

partial void M(IEnumerable<string> s);
partial void M(IEnumerable<string?> s) { } // ok

partial void M(IEnumerable<string?> s);
partial void M(IEnumerable<string> s) { } // warning
```

I chose an analogous behavior on return types in order to be consistent with this.

Reply:
Please consider testing all pairs of:
```C#
#nullable enable
partial object F();
partial object? F();
#nullable disable
partial object F();
partial object? F();
```

---
In reply to: [441084726](https://github.com/dotnet/roslyn/pull/45128#discussion_r441084726) [](ancestors = 441084726)

Reply:
I'm not sure I understand this suggestion. There are only defining declarations in this sample.

Reply:
Sorry for the confusion. I was suggesting pairs of those signatures where one is a partial method declaration and the other is a partial method implementation.

---
In reply to: [441120309](https://github.com/dotnet/roslyn/pull/45128#discussion_r441120309) [](ancestors = 441120309)